main/Repositories/Database/ChatRepository.cs

```diff
@@ -0,0 +1,112 @@
+            CheckEntityAuditing();
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<Chat> GetChatFromParticipantsAsync(List<string> allParticipants)
+        {
+            var a = from c in _context.Chats
+                                .Where(c => c.Participants.Any() &&
+                                         c.Participants.All(p => allParticipants.Contains(p.UserId))
+                                )
+                    select c;
+
+            return await a.FirstOrDefaultAsync();
+        }
+
+        public async Task<IEnumerable<Chat>> GetChatsWithLastMessageAndParticipantsAsync(string userId, int pageSize, DateTime? createdSince)
+        {
+            UserContext userContext = _contextAccessor.HttpContext.GetUserContext();
+            var a = from c in _context.Chats
+                                .Include(c=>c.Messages.OrderByDescending(m => m.CreatedOn).Take(1))
+                                .Include(c=>c.Participants)
+                                .Take(pageSize)
+                                .Where(c=>
+                                    c.Participants.Select(p=>p.UserId).Contains(userContext.UserID)
+                                    && (createdSince.HasValue && c.CreatedOn > createdSince)
+                                    )
+                                .OrderByDescending(c=>c.CreatedOn)
+                                select c;
+            await a.LoadAsync();
+            return a;
+        }
+
+        public List<ChatParticipant> GetParticipantsFromChat(long chatId)
+        {
+            IQueryable<ChatParticipant> a = from p in _context.ChatParticipants
+                                                        .Where(p => p.ChatId == chatId)
+                                            select p;
+
+            return a.ToList();
+        }
+
+        public async Task<List<string>> GetParticipantIdsFromChat(long chatId)
+        {
+            IQueryable<string> a = from p in _context.ChatParticipants
+                                                        .Where(p => p.ChatId == chatId)
+                                            select p.UserId;
+
+            return await a.ToListAsync();
+        }
+
+        private void CheckEntityAuditing()
+        {
+            UserContext userContext = _contextAccessor.HttpContext.GetUserContext();
+            _context.ChangeTracker.CheckAuditing(userContext);
+        }
+
+        public async Task<Chat> GetChatMessagesWithParticipantsDetails(long chatId, int size, DateTime? createdSince)
+        {
+            UserContext userContext = _contextAccessor.HttpContext.GetUserContext();
+            var a = _context.Chats
+                        .Include(c => c.Participants)
+                            .ThenInclude(p => p.User)
+                        .Include(c => c.Messages
+                                        .Take(size)
+                                        .Where(m => m.CreatedOn > createdSince)
```

Comment:
The same question with createdSince here too

Reply:
Added createdSince.HasValue